Bump to .NET 6 stable

mfkl opened the pull request.

mfkl committed:
Bump to .NET 6 stable

mfkl commented:
I'm HTTP 401 to access the CI build logs, but my guess is the samples don't build because of https://github.com/xamarin/Xamarin.Legacy.Sdk/issues/33, or did I mess up somewhere else?

jonathanpeppers commented:
There is some failure in the iOS workload:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
```

We're using Xamarin.Legacy.Sdk + .NET 6 GA, in [AndroidX](https://github.com/xamarin/AndroidX) and Google Play Services bindings...

Maybe there is something I can do to work around the iOS issue.

jonathanpeppers committed:
Bump to windows-2022 image

Context: https://docs.microsoft.com/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml

jonathanpeppers committed:
Update Github actions

* Install .NET 6 GA
* Xcode 13.1 is default now: https://github.com/actions/virtual-environments/issues/4355

jonathanpeppers committed:
Use nuget.org instead of dotnet6 feed

jonathanpeppers committed:
Update HelloMac.csproj

$(SupportedOSPlatformVersion) should be 10.14

jonathanpeppers committed:
Add HACK for NuGet.targets

Fixes:

    D:\a\1\s\samples\Hello\Hello.csproj error MSB4057: The target "_GetRestoreSettingsPerFramework" does not exist in the project.

jonathanpeppers commented on src/Xamarin.Legacy.Sdk/Sdk/Sdk.props:
```diff
@@ -3,5 +3,21 @@
     <XamarinLegacySdk>true</XamarinLegacySdk>
   </PropertyGroup>
+
+  <!--
+    HACK: declare a _GetRestoreSettingsPerFramework target that some versions of NuGet expect to exist.
+    This target should get replaced by NuGet.targets if needed:
+    https://github.com/NuGet/NuGet.Client/blob/914cacea77660b8758014912ee127c82469ac1d9/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L635-L651
+  -->
+  <Target Name="_GetRestoreSettingsPerFramework" Returns="@(_RestoreSettingsPerFramework)">
+    <ItemGroup>
+      <_RestoreSettingsPerFramework Include="$([System.Guid]::NewGuid())">
+        <RestoreAdditionalProjectSources>$(RestoreAdditionalProjectSources)</RestoreAdditionalProjectSources>
+        <RestoreAdditionalProjectFallbackFolders>$(RestoreAdditionalProjectFallbackFolders)</RestoreAdditionalProjectFallbackFolders>
+        <RestoreAdditionalProjectFallbackFoldersExcludes>$(RestoreAdditionalProjectFallbackFoldersExcludes)</RestoreAdditionalProjectFallbackFoldersExcludes>
+      </_RestoreSettingsPerFramework>
+    </ItemGroup>
+  </Target>
```
@mfkl adding this solved one issue, were you getting errors about this target missing?

After adding this, I hit a new missing target, which is too large to replicate in this project:

https://github.com/NuGet/NuGet.Client/blob/914cacea77660b8758014912ee127c82469ac1d9/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L856-L863

I think these were very late changes to improve no-op restore performance in NuGet in .NET 6.

Need to think on this one...

jonathanpeppers replied:
Ok remembering now... This needs to be updated for VS 2022:

https://github.com/xamarin/Xamarin.Legacy.Sdk/blob/4ba5930273eb5bd64b142cfdc574626c68ad41ab/src/Xamarin.Legacy.Sdk/Sdk/Xamarin.VsInstallRoot.targets#L6-L7

jonathanpeppers committed:
Revert "Add HACK for NuGet.targets"

This reverts commit 81dbd0de098db10675c59f5b7948f81dfaaea055.

jonathanpeppers committed:
VS 2022 paths

jonathanpeppers commented:
This is now hitting a new iOS error:
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=15.2.0.17, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' does not have an implementation. [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\Hello\Hello.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: The "GenerateBuildSessionId" task failed unexpectedly. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018: System.TypeLoadException: Method 'Trace' in type 'NullTracer' from assembly 'Xamarin.iOS.Tasks, Version=15.2.0.17, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' does not have an implementation. [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager..ctor() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.BuildTracerManager.get_Instance() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\BuildTracerManager.cs:line 23 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.SetupTracing() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 49 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.MessagingBuildContext.Initialize(String version) in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\MessagingBuildContext.cs:line 15 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId.Execute() in D:\a\_work\1\s\src\MSBuild\Xamarin.Messaging.Build.Client\Tasks\GenerateBuildSessionId.cs:line 26 [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Messaging.Build.targets(79,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\samples\tests\VsInstallRoot\VsInstallRoot.csproj]
```

jonathanpeppers committed:
Install newer Xamarin.VisualStudio.Apple.Sdk.vsix

jonathanpeppers committed:
Revert "Install newer Xamarin.VisualStudio.Apple.Sdk.vsix"

This reverts commit 8176afcf886947a55dfe3363873170eb3ca996dd.

jonathanpeppers committed:
[ios] patch $(MessagingBuildClientAssemblyFile)

This solves an assembly-load error at build time.

mfkl commented:
Good job!
